site/build.rs

```diff
@@ -82,3 +71,23 @@ fn main() -> Result<(), Box<dyn Error>> {
 
     Ok(())
 }
+
+/// If the manifest has `profile.*` entries, read some of the compilation metadata from a built-in
+/// profile with the given `profile` name, if it exists (for example, the options for cargo's
+/// optimized profile is named `release`).
+/// Note that some, or all, of the metadata that we want to display may be missing from the
+/// manifest: it just won't be shown in the UI in that case.
+fn read_profile_metadata(profiles: Option<&toml::Value>, profile: &str) -> ProfileMetadata {
+    let profile = profiles
+        .and_then(|p| p.as_table())
+        .and_then(|t| t.get(profile))
+        .and_then(|t| t.as_table());
+    let debug = profile.and_then(|t| t.get("debug"));
+    let lto = profile.and_then(|t| t.get("lto"));
+    let codegen_units = profile.and_then(|t| t.get("codegen-units"));
+    ProfileMetadata {
+        debug: debug.map(|v| v.to_string()),
+        lto: lto.map(|v| v.to_string()),
+        codegen_units: codegen_units.and_then(|v| v.as_integer().map(|v| v as u32)),
```

Comment:
INB4 we add `codegen-units = "auto"` :laughing: Maybe we should just treat everything as a black box and store it as a string. But it's not super important (and pre-existing of course, I love to nitpick my own old code).

Reply:
We could indeed. I think if we display more metadata in the future, we could simplify this a bit. For now it seems fine, as it's not clear we'll change this code in the short term?

Reply:
Yeah that was written mostly in jest :) Let's keep it as it is for now.